### PR TITLE
feat: display noise detection notification

### PR DESCRIPTION
**Pull request by andrei-gavrilescu**
This depends on https://github.com/jitsi/lib-jitsi-meet/pull/1013.
The PR implements a notification that will pop up when the lib-jitsi-meet conference generates "NOISY_MIC" event.

### CODE REVIEW

File: react/features/noise-detection/middleware.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    const { conference } = action;
+
+    switch (action.type) {
+    case APP_WILL_MOUNT:
+        dispatch(registerSound(NOISY_AUDIO_INPUT_SOUND_ID, NOISY_AUDIO_INPUT_SOUND_FILE));
+        break;
+    case APP_WILL_UNMOUNT:
+        dispatch(unregisterSound(NOISY_AUDIO_INPUT_SOUND_ID));
+        break;
+
```
Comment (hristoterezov) [inline react/features/noise-detection/middleware.js]: style: redundant empty line

File: react/features/noise-detection/middleware.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            });
+        conference.on(
+            JitsiConferenceEvents.NOISY_MIC, () => {
+                const notification = showNotification({
+                    titleKey: 'toolbar.noisyAudioInputTitle',
+                    descriptionKey: 'toolbar.noisyAudioInputDesc'
+                });
+
+                dispatch(notification);
+
```
Comment (hristoterezov) [inline react/features/noise-detection/middleware.js]: style: redundant empty line

File: react/features/noise-detection/middleware.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+// @flow
+
+import { APP_WILL_MOUNT, APP_WILL_UNMOUNT } from '../base/app';
+import { CONFERENCE_JOINED } from '../base/conference';
+import { JitsiConferenceEvents } from '../base/lib-jitsi-meet';
+import { MiddlewareRegistry } from '../base/redux';
+import { playSound, registerSound, unregisterSound } from '../base/sounds';
+import {
+    hideNotification,
+    showNotification
```
[the remark this answers is not in the document]
Comment (andrei-gavrilescu) [reply react/features/noise-detection/middleware.js]: well I didn't do it deliberately :). Probably eslint auto-format mishap. fixed

File: react/features/noise-detection/middleware.js (hunk 1 above)
Comment (andrei-gavrilescu) [reply react/features/noise-detection/middleware.js]: removed.

File: react/features/noise-detection/middleware.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+} from '../notifications';
+
+import { setNoisyAudioInputNotificationUid } from './actions';
+import { NOISY_AUDIO_INPUT_SOUND_ID } from './constants';
+import { NOISY_AUDIO_INPUT_SOUND_FILE } from './sounds';
+
+MiddlewareRegistry.register(store => next => action => {
+    const result = next(action);
+    const { dispatch, getState } = store;
+    const { conference } = action;
```
[the remark this answers is not in the document]
Comment (andrei-gavrilescu) [reply react/features/noise-detection/middleware.js]: So I completely misunderstood the comment. Disregard the paragraph above. I'll add the changes as suggested.

File: react/features/noise-detection/middleware.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import { hideNotification, showNotification } from '../notifications';
+
+import { setNoisyAudioInputNotificationUid } from './actions';
+import { NOISY_AUDIO_INPUT_SOUND_ID } from './constants';
+import { NOISY_AUDIO_INPUT_SOUND_FILE } from './sounds';
+
+MiddlewareRegistry.register(store => next => action => {
+    const result = next(action);
+
+
```
Comment (hristoterezov) [inline react/features/noise-detection/middleware.js]: style: redundant empty line

Comment (andrei-gavrilescu) [reply react/features/noise-detection/middleware.js]: removed.

**Comment by damencho** [comment]
Jenkins add to whitelist

**Pull request merged**